.github/workflows/build-test-deploy.yml

```diff
@@ -1014,63 +1014,93 @@ jobs:
+            echo "::set-output name=hashChange::false"; 
+          else echo "::set-output name=hashChange::true"; 
+          fi
+
+      - uses: cachix/cachix-action@f5f67badd061acb62b5c6e25e763572ca8317004
         with:
-          node-version: 16
-          always-auth: true
-          registry-url: 'https://registry.npmjs.org'
-      - run: yarn install && yarn export --filter=picasso
-        working-directory: frontend
-        env:
-          SUBSTRATE_PROVIDER_URL_KUSAMA_2019: ${{ secrets.SUBSTRATE_PROVIDER_URL_KUSAMA_2019 }}
-          SUBSTRATE_PROVIDER_URL_KUSAMA: ${{ secrets.SUBSTRATE_PROVIDER_URL_KUSAMA }}
-          SUBSTRATE_PROVIDER_URL_KARURA: ${{ secrets.SUBSTRATE_PROVIDER_URL_KARURA }}
-          SUBSQUID_URL: ${{ secrets.SUBSQUID_URL }}
-          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
-      - uses: FirebaseExtended/action-hosting-deploy@0f248036885ae672f78587669fa2923f0ef6cac2
+          skipPush: true
+          installCommand: |
+            nix-channel --add ${{env.NIX_NIXPKGS_CHANNEL }} nixpkgs
+            nix-channel --update
+            nix-env -iA nixpkgs.cachix
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+          name: composable-community
+      - run: |
+          cachix watch-exec --jobs 16 --compression-level 0 composable-community nix -- build .#frontend-static-firebase --no-update-lock-file --show-trace -L
+          mkdir -p firebase/picasso/out/
+          mkdir -p firebase/pablo/out/
+          cp -R result/picasso/* firebase/picasso/out/
+          cp -R result/pablo/* firebase/pablo/out/
+          cp frontend/apps/pablo/.firebaserc firebase/pablo/
+          cp frontend/apps/pablo/firebase.json firebase/pablo/
+          cp frontend/apps/picasso/.firebaserc firebase/picasso/
+          cp frontend/apps/picasso/firebase.json firebase/picasso/
+          nix-env -iA nixpkgs.nodejs
+
+      - name: Deploy Picasso on preview channel
+        if: steps.branch-name.outputs.current_branch != 'main' && steps.feStaticHashChanges.outputs.hashChange == 'true'
+        uses: FirebaseExtended/action-hosting-deploy@0f248036885ae672f78587669fa2923f0ef6cac2
         with:
           repoToken: ${{ secrets.GITHUB_TOKEN }}
           firebaseServiceAccount: ${{ secrets.FIREBASE_SERVICE_ACCOUNT_PICASSO_WEB_APP }}
-          projectId: picasso-test-70bf7
-          entrypoint: frontend/apps/picasso
-
-  deploy-pablo-frontend:
-    name: "Firebase Pablo Web App Deploy"
-    needs:
-      - local-integration-tests
-      - unit-tests
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+          projectId: ${{ secrets.FIREBASE_PICASSO_PROJECT_ID }}
+          entrypoint: firebase/picasso/
+          channelId: ${{ steps.extract_branch.outputs.branch }}
+      - name: Deploy Picasso on live channel
+        if: steps.branch-name.outputs.current_branch == 'main' && steps.feStaticHashChanges.outputs.hashChange == 'true'
```

Comment:
if it is push to main (see other jobs) then it likely should be gated by effects gate (needs) so that only after all tests passed it is deployed if live channel is shared ID. 

or it has new random ID per deploy? if it is random, okey.

Reply:
without effects gate passed, new ui should not be deployed if others can use same dns/host name to access stuff.

Reply:
deploy-devnet:
      name: "Devnet(Rococo Local Dali Dev) Deploy"
      if: github.event_name == 'push' && github.ref_name == 'main'
      needs:
        - effects-gate